tests/urlpatterns/polymorphic_urls.py

```diff
@@ -0,0 +1,32 @@
+from django.urls import path, re_path, register_converter
+
+from . import converters, views
+
+register_converter(converters.DynamicConverter, 'dynamic')
+
+
+# test reverse resolution of routes that share the same name, based on ...
+urlpatterns = [
+    # ... count
+    path('a/1/', views.empty_view, name='a'),
+    path('a/2/<value>/', views.empty_view, name='a'),
+
+    # ... keyword
+    path('b/1/<a>/', views.empty_view, name='b'),
+    path('b/2/<b>/', views.empty_view, name='b'),
+
+    # ... built-in converter type, ordered from general to specific
+    path('c/1/<path:value>/', views.empty_view, name='c'),
+    path('c/2/<str:value>/', views.empty_view, name='c'),
+    path('c/3/<slug:value>/', views.empty_view, name='c'),
+    path('c/4/<int:value>/', views.empty_view, name='c'),
+    path('c/5/<uuid:value>/', views.empty_view, name='c'),
+
+    # ... re_path type
+    re_path(r'^d/1/([A-Z]+)/', views.empty_view, name='d'),
+    re_path(r'^d/2/([a-z]+)/', views.empty_view, name='d'),
+
+    # ... custom converter raising ValueError
+    path('e/1/<value>/', views.empty_view, name='e'),
+    path('e/2/<dynamic:value>/', views.empty_view, name='e'),
+]
```

Comment:
All of these test cases already pass without the change to `resolvers.py`, except for the last one. I added the others to document the various ways Django currently handles polymorphic url name resolution.

Reply:
Extra tests are fine but they should be moved to a separate commit.